system_tests_delivery_counts still failing

reopen of https://github.com/skupperproject/skupper-router/issues/54 (note that here is different test, #54 was `test_06_large_message_released`)

```
41: =================================== FAILURES ===================================
41: __________ OneRouterLinkCountersTest.test_02_large_mesage_presettled ___________
41: 
41: self = <system_tests_delivery_counts.OneRouterLinkCountersTest testMethod=test_02_large_mesage_presettled>
41: 
41:     def test_02_large_mesage_presettled(self):
41: >       self.verify_presettled_count(True)
41: 
41: /home/runner/work/skupper-router/skupper-router/skupper-router/tests/system_tests_delivery_counts.py:1092: 
41: _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
41: /home/runner/work/skupper-router/skupper-router/skupper-router/tests/system_tests_delivery_counts.py:1023: in verify_presettled_count
41:     self.assertEqual(test.sender_stats['deliveryCount'], self.COUNT)
41: E   AssertionError: 36 != 40
```

https://github.com/skupperproject/skupper-router/runs/5488642947?check_suite_focus=true#step:9:4328

--- a/tests/system_tests_delivery_counts.py
+++ b/tests/system_tests_delivery_counts.py
@@ -906,7 +906,8 @@ class OneRouterLinkCountersTest(TestCase):
                     # to propagate the outcome back to the sender the sender's outcomes would lag behind the receivers
                     restart_poll_timer = False
                     self._cleanup()
-            elif self.receiver_stats and self.receiver_stats['deliveryCount'] == self.rx_limit:
+            elif self.receiver_stats and self.receiver_stats['deliveryCount'] == self.rx_limit\
+                    and self.sender_stats and self.sender_stats['deliveryCount'] == self.count:
                 restart_poll_timer = False
                 self._cleanup()
 

